src/Http/Http/test/StreamPipeReaderTests.cs

```diff
@@ -477,6 +477,42 @@ public async Task ConsumePartialBufferBetweenMultipleSegmentsWorks()
             Reader.AdvanceTo(readResult.Buffer.End);
         }
 
+        [Fact]
+        public async Task SetMinimumReadThresholdSegmentAdvancesCorrectly()
+        {
+            Reader = new StreamPipeReader(MemoryStream,
+                minimumSegmentSize: 16,
+                minimumReadThreshold: 8,
+                new TestMemoryPool());
+
+            Write(Encoding.ASCII.GetBytes(new string('a', 8)));
+            var readResult = await Reader.ReadAsync();
+            Reader.AdvanceTo(readResult.Buffer.Start, readResult.Buffer.End);
+
+            Write(Encoding.ASCII.GetBytes(new string('a', 16)));
+            readResult = await Reader.ReadAsync();
+
+            Assert.False(readResult.Buffer.IsSingleSegment);
+        }
+
+        [Fact]
+        public async Task SetMinimumReadThresholdToMiminumSegmentSizeAlwaysGetsNewBlock()
+        {
+            // Every call to ReadAsync will always get a new block (even if nothing was read).
```

Comment:
> Every call to ReadAsync will always get a new block (even if nothing was read).

Why?

Reply:
The threshold is equal to the minimum block size (which in our case, is always the block size). After calling ReadAsync a second time, the next read will be below the threshold and create a new block.

Reply:
Why not? I don't understand why you'd suddenly stop ensuring a minimum read buffer size just because there's examined-but-not-consumed bytes from the last read.